_posts/2023-02-26-pernStack.md

```diff
@@ -0,0 +1,36 @@
+
+## Why we choose PERN Stack
+
+PERN is an acronym for PostgreSQL, Express, React and Node.
+[![](https://mermaid.ink/img/pako:eNpNTU1LxDAQ_SthTgrd0iRN0w2yh0WvIuhJchnadD9okpKkuGvpfzeriDKHeW_exyzQ-d6AgmH0H90RQyJve-2CwS6V50geNpsdebpMwcSYuXZ_-E7DP0E78pybbhDuf2IvPqZD1rW7DRRgTbB46vO3JduJhnQ01mhQGfZmwHlMGrRbsxXn5F-vrgOVwmwKmKcek3k84SGgBTXgGPN1Qvfuvf01ZQpqgQsoyUveUCmYYFII3vICrqBqVm4lk4wyUefNmVgL-PwuqMqGVU3D2i2tWdVWlK5fdjZYXA?type=png)](https://mermaid.live/edit#pako:eNpNTU1LxDAQ_SthTgrd0iRN0w2yh0WvIuhJchnadD9okpKkuGvpfzeriDKHeW_exyzQ-d6AgmH0H90RQyJve-2CwS6V50geNpsdebpMwcSYuXZ_-E7DP0E78pybbhDuf2IvPqZD1rW7DRRgTbB46vO3JduJhnQ01mhQGfZmwHlMGrRbsxXn5F-vrgOVwmwKmKcek3k84SGgBTXgGPN1Qvfuvf01ZQpqgQsoyUveUCmYYFII3vICrqBqVm4lk4wyUefNmVgL-PwuqMqGVU3D2i2tWdVWlK5fdjZYXA)
+
+PostgreSQL - It is an Object-Relational Database management system that uses the SQL language combined with many other features that safely store and scale data.
+
+Express - It is a back-end framework for Node.js. used for building web applications and specially APIs. It provides a thin layer of fundamental web application features.Express makes back-end code easier and simpler to write.Creating efficient and robust API is quick and easy.
+Allows you to define an error handling middleware.
+
+React - A Front end Framework teaching at Turing. On eo f the requirement for the project is to use react framework. Couples of benefit are Virtual DOM in ReactJS makes user experience better and developer’s work faster. It guarantees stable code.
+React allows its components to be reused saving time and effort. Provide the feature of Virtual DOM.
+
+Node.js is Javascript runtime environment for developing server-side and networking applications.
+
+- ### How We go about setting up a workflow
+  We approach our workflow from FE perspective where we consider user story first to design how our data table is going to be. First, we wrote the MVP UI, and create the server using Express. Then, we connect express server to postgres. Once the two framework connect, we seed our mock data to the database and voila. Time to deployed!
+- ### What else we added to out PERN stack to make it work
+  We need Knex to create schema and write query to postgresSQL and we need to add docker compose and also table plus for the ease of interacting with database.
+- ### What are our blockers how did we overcome it
+- Ops for library than writing code from scratch.
+  In our UI, we wanted to have a robust filter function that can filter multiple keywords and also has interactive features. As this function evolved, the code got messier and buggier to a point that we decided use a filter library instead. Our reasoning was, we want to get to the core of the project which is building BE and using a library not only save us time but also, creating more stable feature.
+- Changing MVP decision
```

Comment:
You need a different title for this bullet point... the other two are more about the lessons:
* Opt for using a library instead of writing code from scratch
* Asking for help

This one needs to be different because it doesn't really fit the same model as the others.